ipv-core/src/main/java/com/ibm/whc/deid/providers/identifiers/DateTimeIdentifier.java

```diff
@@ -31,120 +24,115 @@
 public class DateTimeIdentifier extends AbstractIdentifier {
   private static final long serialVersionUID = 2544117023916280649L;
 
+  private static final LogManager log = LogManager.getInstance();
+
   private static final String[] appropriateNames =
       {"Datetime", "Timestamp", "Birthday", "Birth date", "Date", "BirthDate", "Date of birth"};
 
   private static final String[] patterns = {"dd-MM-yyyy", "dd-MMM-yyyy", "yyyy-MM-dd", "dd/MM/yyyy",
       "yyyy/MM/dd", "dd-MM-yyyy[ HH:mm:ss]", "yyyy-MM-dd[ HH:mm:ss]", "dd/MM/yyyy[ HH:mm:ss]",
       "yyyy/MM/dd[ HH:mm:ss]"};
 
+  // These patterns are used as a "fast-check" for each of the datetime patterns above.
   private static final Pattern[] datePatterns =
       new Pattern[] {Pattern.compile("^\\d{2}-\\d{2}-\\d{4}$"),
-          Pattern.compile("^\\d{2}-\\w{3}-\\d{4}$"), Pattern.compile("^\\d{4}-\\d{2}-\\d{2}$"),
+          Pattern.compile("^\\d{2}-.{3,}-\\d{4}$"), Pattern.compile("^\\d{4}-\\d{2}-\\d{2}$"),
           Pattern.compile("^\\d{2}/\\d{2}/\\d{4}$"), Pattern.compile("^\\d{4}/\\d{2}/\\d{2}$"),
           Pattern.compile("^\\d{2}-\\d{2}-\\d{4}( \\d{2}:\\d{2}:\\d{2})?$"),
           Pattern.compile("^\\d{4}-\\d{2}-\\d{2}( \\d{2}:\\d{2}:\\d{2})?$"),
           Pattern.compile("^\\d{2}/\\d{2}/\\d{4}( \\d{2}:\\d{2}:\\d{2})?$"),
           Pattern.compile("^\\d{4}/\\d{2}/\\d{2}( \\d{2}:\\d{2}:\\d{2})?$")};
 
-  private static final DateTimeFormatter dateFormats[] = new DateTimeFormatter[patterns.length + 1];
-
-  private static final LogManager log = LogManager.getInstance();
+  // Indicates whether the pattern includes an alphabetic component that could benefit from
+  // character case awareness
+  private static final boolean[] patternHasVariableCaseComponent =
+      {false, true, false, false, false, false, false, false, false};
 
+  private static final DateTimeFormatter dateFormats[] = new DateTimeFormatter[patterns.length];
   static {
-    dateFormats[patterns.length] = DateTimeFormatter.ISO_INSTANT;
     for (int i = 0; i < patterns.length; i++) {
-      dateFormats[i] = new DateTimeFormatterBuilder().appendPattern((patterns[i]))
+      dateFormats[i] = new DateTimeFormatterBuilder().parseCaseInsensitive()
+          .appendPattern((patterns[i]))
           .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
           .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
-          .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0).toFormatter().withZone(ZoneOffset.UTC);
+          .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0).toFormatter();

```

Comment:
Are we going to support the time zone info somewhere else instead ? It's mandated in healthcare scenarios. We had SOP courses about this in good old GxP where they forced that offset to prevent time zone mistakes

Reply:
These changes are not dropping timezone name of offset awareness.  They are correcting errors where timezone offsets were assumed.  If the input includes offset or time zone information, that information is retained and is used when the output is written, when appropriate.  Comparisons to the current date and time take into account whether the input has offset or timezone information and if so, the current date and time is obtained in a manner that includes similar information for comparison purposes.